guides/release/testing/different-types-of-tests.md

```diff
@@ -0,0 +1,221 @@
+- Check how [`normalize()`](https://api.emberjs.com/ember-data/release/classes/JSONAPISerializer/methods/normalize?anchor=normalize) in a serializer receives data
+- Check how [`serialize()`](https://api.emberjs.com/ember-data/release/classes/JSONAPISerializer/methods/serialize?anchor=serialize) in a serializer sends data
+- A [cron](https://en.wikipedia.org/wiki/Cron) utility parses an input string into an object that can be used for UI
+
+### What to Watch Out for
+
+When unit tests involve the Ember framework, you must import and call [`setupTest()`](https://github.com/emberjs/ember-qunit#setup-tests), then pass the `hooks` object. (Don't worry. [Ember CLI](../#toc_ember-cli) will do this for you!)
+
+For example, consider a service that keeps an array of messages, to be shown to the user at a later time:
+
+```javascript {data-filename=tests/unit/services/flash-messages-test.js}
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+
+module('Unit | Service | flash-messages', function(hooks) {
+  setupTest(hooks);
+
+  test('should be able to buffer messages', function(assert) {
+    let service = this.owner.lookup('service:flash-messages');
+
+    service.add('Hello');
+    service.add('World!');
+
+    assert.deepEqual(service.get('messages'), ['Hello', 'World!']);
+  });
+});
+```
+
+By calling `setupTest()`, you gain access to a few things. First is Ember's [Dependency Injection](../../applications/dependency-injection/) system. In short, you can [look up](https://emberjs.com/api/ember/release/classes/ApplicationInstance/methods/lookup?anchor=lookup) anything in your application, with a little help from `this.owner`. Second, you gain access to some common utility functions, `this.get()` and `this.set()`, in your tests. Finally, you can use `pauseTest()` to [debug your tests](../#toc_how-to-debug-tests).
+
+
+## Rendering Tests
+
+### Definition
+
+Rendering tests (previously known as integration tests) check how a **component looks and behaves**. Ember CLI creates rendering tests for [components](../../components/defining-a-component/) and [helpers](../../templates/writing-helpers/).
+
+In terms of performance, rendering tests sit in the middle, between unit and application tests.
+
+### Why Use Them?
+
+Since your app is made up of multiple components, you want to ensure that each is correct before testing them as a group. If a component is reusable, you want to guarantee that it works for all (if not, many) permutations of [arguments](../../components/arguments-and-attributes/) and [actions](../../components/actions-and-events/#toc_passing-down-the-action).
+
+Rendering tests also let you test components in the context of an Ember application. When you use `render` to create a component, you are guaranteed that the component will follow Ember's lifecycle hooks. You can use the component lifecycle to trace errors in components and rendering tests.
```

Comment:
Yeah, I see what you mean about "context" being misleading. I think your suggestion of "exercises the same code [...]"—while it gets to the point, which I like—may be a bit vague.

Maybe, combining both of our ideas:

"Rendering tests let you test components using Ember's rendering engine. This means, a component created in your rendering test will behave as it would in the real app. You are guaranteed that the component will follow Ember's lifecycle hooks. You can also interact with the component like an end-user would."